doc/src/tutorials/physics/mechanics/duffing-example.rst

```diff
@@ -0,0 +1,146 @@
+.. _duffing-example:
+
+==================================
+Duffing Oscillator with a Pendulum
+==================================
+
+In this example we demonstrate the use of functionality provided in
+:obj:`sympy.physics.mechanics` for deriving the quations of motion (EOM) for a system
+consisting of a Duffing oscillator with a pendulum. This example is inspired by the
+paper [P.Brzeskia2012]_ section 2.
+
+.. _fig-duffing-oscillator-pendulum:
+.. figure:: duffing-oscillator-pendulum.svg
+
+The system will be modeled using Lagrange equations. `M` is mass of the Duffing oscillator,
+`m` is mass of the pendulum, `l` is length of the pendulum. `k_1` and `k_2` are linear and
+non-linear parts of spring stiffness, and `c_1` is a viscous damping coefficient of the Duffing oscillator.
+
+   >>> import sympy as sm
+   >>> import sympy.physics.mechanics as me
+   >>> me.init_vprinting()
+
+Define Variables
+================
+
+   >>> M, m, l, k1, k2, c1, g, h, w, d, r = sm.symbols('M, m, l, k1, k2, c1, g, h, w, d, r')
+   >>> q1, q2, u2, u2 = me.dynamicsymbols('q1 q2 u1 u2')
+   >>> q1d = me.dynamicsymbols('q1', 1)
+
+- :math:`h`: Height of the Duffing oscillator
+- :math:`w`: Width of the Duffing oscillator
+- :math:`d`: Depth of the Duffing oscillator
+- :math:`r`: Radius of the massive bob of the pendulum
+- :math:`q_1`: Generalized coordinate representing the position of the Duffing oscillator
+- :math:`q_2`: Generalized coordinate representing the angle of the pendulum
+- :math:`u_1`: Generalized speed associated with the Duffing oscillator
+- :math:`u_2`: Generalized speed associated with the pendulum
+
+Define Kinematics
+=================
+
+Define all the reference frames and points.
+
+   >>> N = me.ReferenceFrame('N')
+   >>> B = N.orientnew('B', 'axis', (q2, N.z))
+
+The angular velocity of the pendulum in the reference frame is:
+
+   >>> B.ang_vel_in(N)
+   q2'(t) n_z
+
+Locations and velocities of the Duffing Oscillator block and the pendulum are:
+
+   >>> O = me.Point('O')
+   >>> block_point = O.locatenew('block', q1 * N.y)
+   >>> pendulum_point = block_point.locatenew('pendulum', l * B.y)
+
+O is a fixed point in the inertial reference frame.
+
+   >>> O.set_vel(N, 0)
+   >>> block_point.set_vel(N, q1d * N.y)
+   >>> pendulum_point.v2pt_theory(block_point, N, B)
+   q1'(t) n_y + -l*q2'(t) b_x
+
+Define inertia and rigid bodies.
+Here, we assume a simple pendulum which consists of a bob of mass m hanging from a massless string of length l
+and fixed at a pivot point (Duffing Oscillator Block).
+
+   >>> I_block = me.inertia(N, M*(h**2 + d**2)/12, M*(w**2 + h**2)/12, M*(w**2 + d**2)/12)
+   >>> I_pendulum = me.inertia(B, 0, 0, 2*m*r**2/5)
+
+   >>> par_block = me.RigidBody('block', block_point, N, M, (I_block, block_point))
+   >>> par_pendulum = me.RigidBody('pendulum', pendulum_point, B, m, (I_pendulum, pendulum_point))
```

Comment:
Maybe rename to `block_body` or `block_rb` and `pendulum_...`, as you are modeling them as rigid bodies and not as particles.